### DISCUSSION TITLE
How to run the compiled file

### DISCUSSION THREAD

**Discussion by Yasqader** [abstools/abstools#329] [Q&A] [answered]
Hello everyone;

During compilation, model.jar file should be generated but i don't have this jar file in the path. I'll be very grateful if someone can help me to resolve this problem.

and also i have ' run ' file in erl path but it didn't run and the message  " Command 'run' not found " is shown.

**Comment by rudi** [comment]
Hi, please show the exact shell commands that you type to compile and start the model, and the output from the compiler

**Comment by Yasqader** [comment]
$ absc --erlang *.abs                                // while run is generated and located in gen/erl the message " Command 'run' not found " is shown
  $ gen/erl/run

And
  $ absc --java *.abs -o model.jar
  $ java -jar model.jar                               //   no jar file is generated and located in the path 
[SmartHome.zip](https://github.com/abstools/abstools/files/11361433/SmartHome.zip)
[erl.zip](https://github.com/abstools/abstools/files/11361435/erl.zip)

**Comment by Yasqader** [comment]
And also the ABS folder is generated with its libraries in the java mode.Result is the same both in compile using EMACS and in the Terminal using absc, no message is shown in terminal and in EMACS **compilation finished** is shown. but at the time of execution the compiled files (after java -jar model.jar), **" Error: Unable to access jarfile model.jar "** is shown in the terminal. Folders generated after compile was sent here before **SmartHome.zip** & **erl.zip** .

**Comment by rudi** [comment, marked as answer]
This is a bit confusing.  Here's a short abs program `hello.abs`:

```
module Hello;
{
    println("Hello world!");
}
```

Here's a transcript of me running it on the Java and Erlang backends:

```
~/Source/tmp% rm -rf gen
~/Source/tmp% absc --version
ABS Tool Suite version HEAD
Built from git tree v1.9.3-200-ge83d114cf
~/Source/tmp% absc --erlang hello.abs
~/Source/tmp% gen/erl/run
Hello world!
~/Source/tmp% absc --java hello.abs -o hello.jar
~/Source/tmp% java -jar hello.jar
Hello world!
~/Source/tmp%
```

Could you paste a transcript of the same commands on your machine?  Also, what operating system do you use?

**Reply by Yasqader** [reply]
First i should say your cooperation is very valuable, thank you very much.

And

Linux ubuntu is installed on the machine and here is the transcript.

```
~$ rm -rf gen
~$ absc --version
ABS Tool Suite version HEAD
Built from git tree v1.9.3-172-ge720b8ece
~$ absc --erlang Hello.abs
~$ gen/erl/run
Hello world!
~$ absc --java Hello.abs -o Hello.jar
~$ java -jar Hello.jar
Error: Unable to access jarfile Hello.jar
```
Using println on main block, the code is executable on erlang now, but not with println on deltas, in fact the code is executed but print nothing and the problem with java remained, no jar file is generated.

Indeed i have another problem with chat code in the DSPL mode based on the article **"Executable Modelling of Dynamic Software Product Lines in the ABS Language"**,  according to the article using **stateupdate** the product can be changed on runtime but stateupdate is not a keyword in the ABS, can you take a look on the code? if yes where is better to send the code,  here in discussions or in other part of the repository?

**Reply by rudi** [reply]
Your abs version is 28 commits behind master, so I think it simply does not include the feature you want to use (generation of a jar file from the Java backend).

**Reply by rudi** [reply]
Re your other question, there are academic articles that describe versions of abs that either were never implemented, never merged to master, or never added documentation and unit tests, so the functionality they described might have bit-rotted.  You might want to contact the authors and enquire of the status of the code they presented in the paper.

**Reply by Yasqader** [reply]
Thank you for advise.
